tests/product/test_connectors.py

```diff
@@ -61,6 +61,96 @@ def test_basic_connector_add_remove(self):
+        self.setup_cluster(NoHadoopBareImageProvider(),
+                           self.STANDALONE_PRESTO_CLUSTER)
+        self.run_prestoadmin('server start')
+        for host in self.cluster.all_hosts():
+            self.assert_has_default_connector(host)
+
+        self._assert_connectors_loaded([['system'], ['tpch']])
+
+        self.run_prestoadmin('connector remove tpch -H %(master)s')
+        self.run_prestoadmin('server restart')
+        self.assert_path_removed(self.cluster.master,
+                                 os.path.join(constants.CONNECTORS_DIR,
+                                              'tpch.properties'))
+        self._assert_connectors_loaded([['system']])
+        self.assert_path_removed(self.cluster.master,
+                                 os.path.join(constants.REMOTE_CATALOG_DIR,
+                                              'tpch.properties'))
+        for slave in self.cluster.slaves:
+            self.assert_has_default_connector(slave)
+
+        self.cluster.write_content_to_host(
+            'connector.name=tpch',
+            os.path.join(constants.CONNECTORS_DIR, 'tpch.properties'),
+            self.cluster.master
+        )
+        self.run_prestoadmin('connector add tpch -H %(master)s')
+        self.run_prestoadmin('server restart')
+        for host in self.cluster.all_hosts():
+            self.assert_has_default_connector(host)
+        self._assert_connectors_loaded([['system'], ['tpch']])
+
+    def test_connector_add_remove_worker_using_dash_h(self):
+        self.setup_cluster(NoHadoopBareImageProvider(),
+                           self.STANDALONE_PRESTO_CLUSTER)
+        self.run_prestoadmin('server start')
+        for host in self.cluster.all_hosts():
+            self.assert_has_default_connector(host)
+
+        self._assert_connectors_loaded([['system'], ['tpch']])
+
+        self.run_prestoadmin('connector remove tpch -H %(slave1)s')
+        self.run_prestoadmin('server restart')
+        self._assert_connectors_loaded([['system'], ['tpch']])
+        self.assert_path_removed(self.cluster.slaves[0],
+                                 os.path.join(constants.REMOTE_CATALOG_DIR,
+                                              'tpch.properties'))
+        self.assert_has_default_connector(self.cluster.master)
+        self.assert_has_default_connector(self.cluster.slaves[1])
+        self.assert_has_default_connector(self.cluster.slaves[2])
+
+        self.cluster.write_content_to_host(
+            'connector.name=tpch',
+            os.path.join(constants.CONNECTORS_DIR, 'tpch.properties'),
+            self.cluster.master
+        )
+        self.run_prestoadmin('connector add tpch -H %(slave1)s')
+        self.run_prestoadmin('server restart')
+        self.assert_has_default_connector(self.cluster.slaves[0])
+        self._assert_connectors_loaded([['system'], ['tpch']])
+
+    def test_connector_add_remove_two_workers_using_dash_h(self):
```

Comment:
Agree wholeheartedly. The product tests are a very expensive way to find bugs. This tests the behavior of the command line parsing more than the actual product tests.